File: homework/12_B/02_Aleksandar/homework1/circle.py
from enum import Enum
from math import sqrt

class Intersection(Enum):
    TOUCHING = 1
    NOTTOUCHING = 2
    INTERSECT = 3
    SAME = 4
    INSIDE = 5
    INVALID = 6

class Circle:
  def __init__(self, x, y,radius):
    self.x = x 
    self.y = y
    self.radius = radius

  def check(self,other):    
    dist = sqrt((self.x-other.x)**2 + (self.y-other.y)**2)
    print (self.x)
    print (other.x)
    print (self.y)
    print (other.y)

    if self.radius <= 0 or other.radius <= 0:
        return Intersection.INVALID.name # INVALID
    if self.x == other.x and self.y == other.y and self.radius == other.radius:
        return Intersection.SAME.name # SAME
    elif (dist <= (self.radius - other.radius) or dist <= (other.radius - self.radius)):
        return Intersection.INSIDE.name # INSIDE
    elif dist == (self.radius + other.radius):
        return Intersection.TOUCHING.name # TOUCHING
    elif dist < (self.radius + other.radius):
        return Intersection.INTERSECT.name # INTERSECT
    else:
        return Intersection.NOTTOUCHING.name # NOTTOUCHING

File: homework/12_B/05_georgi/circle_intersection/src/Circle.java
import java.util.Objects;

import static java.lang.Math.pow;
import static java.lang.Math.sqrt;

public class Circle {
    //centre
    Double x;
    Double y;
    //radius
    Double r;
    String result;
    Double d;

    public Circle(Double x, Double y, Double r) {
        this.x = x;
        this.y = y;
        this.r = r;
    }

    public Double getX() {
        return x;
    }

    public Double getY() {
        return y;
    }

    public Double getR() {
        return r;
    }

    String circle_intersection(Circle circle)
    {
        d = sqrt(pow(circle.getX() - this.x, 2) + pow(circle.getY() - this.y, 2));
        //intersects
        if(d == this.r + circle.getR())
        {
            result = "Circles intersect at one point.";
        }
        else if(this.r - circle.getR() < d && d < this.r + circle.getR())
        {
            result = "Circles intersect at two points.";
        }
        else if(d == this.r || d == circle.getR())
        {
            result = "The centre of one of the circles lies on the other circle.";
        }
        //number of intersections

        //the clash
        else if(Objects.equals(this.x, circle.getX()) && Objects.equals(this.y, circle.getY()) && Objects.equals(this.r, circle.getR()))
        {
            result = "The circles are the same.";
        }

        else
        {
            result = "The circles do not intersect.";
        }
        System.out.println(result);
        return result;
    }
}
